apps/qubit/modules/settings/actions/ldapAction.class.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * LDAP settings
+ *
+ * @package    AccesstoMemory
+ * @subpackage settings
+ */
+
+class SettingsLdapAction extends sfAction
+{
+  public function execute($request)
+  {
+    $this->ldapForm = new SettingsLdapForm;
+
+    // Handle POST data (form submit)
+    if ($request->isMethod('post'))
+    {
+      QubitCache::getInstance()->removePattern('settings:i18n:*');
+
+      // Handle LDAP form submission
+      if (null !== $request->ldap)
+      {
+        $this->ldapForm->bind($request->ldap);
+        if ($this->ldapForm->isValid())
+        {
+          // Do update and redirect to avoid repeat submit wackiness
+          $this->updateLdapSettings();
+          $this->redirect('settings/ldap');
+        }
+      }
+    }
+
+    $this->populateLdapForm();
+  }
+
+  /**
+   * Populate the LDAP form
+   */
+  protected function populateLdapForm()
+  {
+    $this->ldapForm->setDefaults(array(
+      'host' => QubitSetting::getByName('ldapHost')->getValue(array('sourceCulture'=>true)),
+      'port' => QubitSetting::getByName('ldapPort')->getValue(array('sourceCulture'=>true)),
+      'base_dn' => QubitSetting::getByName('ldapBaseDn')->getValue(array('sourceCulture'=>true))
+    ));
+  }
+
+  /**
+   * Update the LDAP settings
+   */
+  protected function updateLdapSettings()
+  {
+    $thisForm = $this->ldapForm;
+
+    $fieldToSettingMap = array(
+      'host'    => 'ldapHost',
+      'port'    => 'ldapPort',
+      'base_dn' => 'ldapBaseDn'
+    );
+
+    foreach($fieldToSettingMap as $field => $setting)
+    {
+      if (null !== $value = $thisForm->getValue($field))
+      {
+        $setting = QubitSetting::getByName($setting);
+        $setting->setValue($value, array('sourceCulture' => true));
+        $setting->save();
+      }
+    }
```

Comment:
How would you feel about not adding these three settings in the fixtures / migration and just populate them when they are created for the first time? I've done that before with other settings page. It saves us the migration, which is always a good thing. Your code above would be almost the same, just consider the possibility that `getByName` returns `NULL` so you have to create a new setting from scratch?

It may be a dirty trick, you don't have to do it if you don't want!